### PR TITLE
fix: Returns a warning instead of panicking if an AppImage is not mounted, closes #7736

### LINKED ISSUE #7736: [bug] AppImage fails to run due to misguided "validation"
### Describe the bug

https://github.com/tauri-apps/tauri/commit/6fbd6dba5290dc017ab0ba5a44cf4358b022836f ("validate AppImage execution when env vars are set") introduces a "validation" that makes unreasonable assumptions, and as a result the AppImage does not work as it should:

```
thread 'main' panicked at '`APPDIR` or `APPIMAGE` environment variable found but this application was not detected as an AppImage; this might be a security issue.', (...)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

References:
* https://github.com/tauri-apps/tauri/issues/7046
* https://github.com/AppImage/appimage.github.io/pull/3242
* https://github.com/AppImage/appimage.github.io/pull/3263
* https://github.com/lencx/ChatGPT/issues/308
* https://github.com/spyglass-search/spyglass/issues/491
* https://github.com/OzymandiasTheGreat/emoji-keyboard/issues/69
* https://github.com/tauri-apps/tauri/issues/5637

### Reproduction

* Try to run the AppImage using [Firejail](https://firejail.wordpress.com/documentation-2/appimage-support/)
* Extract the AppImage to an AppDir using `--appimage-extract`, then run it using `./squashfs-root/AppRun`

### Expected behavior

The application runs.

### Platform and versions

```text
I can't run commands; the result is from an automated test that can be seen at the URL below.
```

### Stack trace

```text
https://github.com/AppImage/appimage.github.io/actions/runs/6060923178/job/16445575611#step:4:386
```

### Additional context

Nothing in the AppImageSpec suggests or guarantees that an AppImage gets mounted to a mountpoint of any sort of predictable path. Just because one particular version of the AppImage runtime had a certain behavior it cannot be deduced that this is universally true; e.g., when using Firejail it is not.

### PR DESCRIPTION
**Pull request by Benji377**
### What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [X] No

### Checklist
- [X] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary

### Other information
When uploading an AppImage to the [AppImage repository](https://github.com/AppImage), Tauri checks if the AppImage is valid by trying to mount it and check a specific mount point. Unfortunately, many AppImages that are perfectly functional don't pass this test, and therefore the automatic AppImage Workflow fails.
To fix this issue, I simply replaced the panic with a warning. This would still give feedback, but at least let the AppImage finish running. I encountered this issue [here](https://github.com/AppImage/appimage.github.io/pull/3263) and mayn other users have as well in the issue #7736 

Basically, I am reverting some changes of the commit: https://github.com/tauri-apps/tauri/commit/6fbd6dba5290dc017ab0ba5a44cf4358b022836f

**Note:** I used the log crate to display a warning, but if that's not ok with you, you are more than welcome to change that.

### CODE REVIEW

**Comment by probonopd** [comment]
> When uploading an AppImage to the [AppImage repository](https://github.com/AppImage), Tauri checks if the AppImage is valid by trying to mount it and check a specific mount point.

...close ;-) This is what actually happens: 

When submitting an AppImage to https://appimage.github.io/, then the automated test in that repository checks if the AppImage is valid by trying to run it with [Firejail](https://github.com/netblue30/firejail).

And since Tauri makes some assumptions about AppImages that do not hold true (and are not part of the spec), Firejail currently fails to run AppImages made with Tauri, preventing Tauri-based AppImages from passing the automated test.

So it would be best to entirely reverse-apply https://github.com/tauri-apps/tauri/commit/6fbd6dba5290dc017ab0ba5a44cf4358b022836f since its assumptions are flawed to begin with.

**Review by lucasfernog-crabnebula — APPROVED** [review]

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]
